### ISSUE TITLE
Media Library fails "cannot read properties of null (reading 'includes')

### ISSUE THREAD

**Issue by shibacomputer** [strapi/strapi#16005]
### Required System information

- Node.js version: 16.19.0
- NPM version: 8.19.3
- Strapi version: 4.6.1
- Database: Sqlite
- Operating system: Debian 5.10.92-1
- Is your project Javascript or Typescript: Javascript

### Describe the bug
Similar to #12761 clicking the media library crashes all of Strapi's front end and outputs the following errors:

```
TypeError: Cannot read properties of null (reading 'includes')
    at le (main.890a972d.js:1376:839)
    at Zl (main.890a972d.js:1744:54681)
    at Hc (main.890a972d.js:1748:9133)
    at Bc (main.890a972d.js:1748:989)
    at qc (main.890a972d.js:1748:917)
    at cu (main.890a972d.js:1748:768)
    at Iu (main.890a972d.js:1746:11068)
    at main.890a972d.js:1744:39503
    at n.unstable_runWithPriority (main.890a972d.js:1776:3795)
    at ta (main.890a972d.js:1744:39281)
```
This makes Strapi completely unusable.

### Steps to reproduce the behavior

1. Log in to Strapi.
2. Upload some media files to the Media Library.
3. Come back later and click the Media Library in the menu.
4. See a quick flash of the Media Library UI (without any user-provided content in it).
5. Crash to a white screen.

### Expected behavior

A working Media Library and some resilience on the front end to prevent user confusion.

### Screenshots

![image](https://user-images.githubusercontent.com/737950/223137684-376b5bac-6b5a-4c41-8de5-7db325bbc86b.png)
![image](https://user-images.githubusercontent.com/737950/223137719-5b93eefb-d094-43f0-be29-218f9e2be9fb.png)

**Comment by shibacomputer** [comment]
Where can I get support for this to at least discover and remove the offending upload? I've never encountered a CMS Media Uploader capable of crashing and burning like this, and as a result my client is unable to launch their project.

**Comment by shibacomputer** [comment]
Additional detail, Strapi is generating hundreds of `.false` files in the `public/uploads` directory. eg:
```
-rw-r--r-- 1 root      root        57818 Mar  4 08:24 thumbnail_W_Xh_A1_WM_0yiy3_A_Tf_YP_8n_FOWFO_5_T3zdvmww_RST_8115_Cb_Gj_Rr_B_Gh_So4kl7_N5ol_Rdivt_YRA_6_HH_Mg_MX_Dn_UJF_Oz2z_R_Po9jq3_Dl_F9_Kt4yh_H_Yd_ZGPZ_6_Tw_GX_1_R_Ky8_O7i1k56_K4z2z7_e2_Jh_VI_6_Xd_CC_Ni16q6_U_8e65a3f848.false
-rw-r--r-- 1 root      root        57818 Mar  4 08:25 thumbnail_W_Xh_A1_WM_0yiy3_A_Tf_YP_8n_FOWFO_5_T3zdvmww_RST_8115_Cb_Gj_Rr_B_Gh_So4kl7_N5ol_Rdivt_YRA_6_HH_Mg_MX_Dn_UJF_Oz2z_R_Po9jq3_Dl_F9_Kt4yh_H_Yd_ZGPZ_6_Tw_GX_1_R_Ky8_O7i1k56_K4z2z7_e2_Jh_VI_6_Xd_CC_Ni16q6_U_9b05601e11.false
```

**Comment by shibacomputer** [comment]
If anyone encounters this issue with the media library, the short term solution is:

1. Take your Strapi CMS offline and back it up.
2. Manually log into your CMS db
3. Find all instances of `mime` `NULL`.

You will also likely find image files with a `.false` file extension. These uploads and their subsequent failed inclusion in the db is what kills the web ui (and in our case, the entire Strapi/Gatsby build). In our case, we replaced empty `mime` fields with `image/jpeg` to get online quickly, and now we have a working web UI and a newly invented image format:
![image](https://user-images.githubusercontent.com/737950/223162769-96b332c4-9362-4753-a6c6-b4c49a71713b.png)

**Comment by Marc-Roig** [comment]
Hello :) Thank you for providing that detailed description, could you share one of the images you used that caused the issue?

**Comment by joshuaellis** [comment]
Hey @shibacomputer i'm sorry you're experiencing these issues. We obviously thought the issue was solved in #12761 and you've since proven this was wrong.

To resolve the issue we're gonna do the following steps:
* I'm gonna patch the FE with checks against null / undefined values in this space to stop the FE from crashing (I'll work towards getting this released tomorrow, but I can't promise this.)
* We'd really appreciate it if you could share one of these assets that causes `null` in the database – if you don't want to share here, I can give you my email.
* From there I can work with @Marc-Roig  to determine how this happening and what parts of the code need to change to stop this from happening in the future this is the root of the problem.

I've also updated the severity to highlight we're taking this seriously for you.

**Comment by shibacomputer** [comment]
Hey @joshuaellis and @Marc-Roig, thanks for the quick response. My terse language is from our own missed deadline as a result of this issue, having encountered #12761 with our client, migrated, then encountering it again.

I've made contact with the client to get permission to share an example asset with you. Can you provide me with a way to securely share this with you?

**Comment by joshuaellis** [comment]
You can send it to my work email josh.ellis@strapi.io however you prefer e.g via email / weTransfer etc.?

**Comment by shibacomputer** [comment]
Sure thing -- once I get the OK from the client I'll forward some examples to you. Hopefully soon!

**Comment by joshuaellis** [comment]
Reopening this issue so we can track the root cause using asset provided – this will auto-close in two weeks if we can't reproduce

**Comment by shibacomputer** [comment]
Hi @joshuaellis, I've just emailed you example images that have caused us issues.

**Comment by joshuaellis** [comment]
Hey @shibacomputer we've reviewed your images and could not reproduce the issues you were facing with these assets.

We're going to add a default mime type when it cannot be calculated in the service to ensure that no null values are passed to the database that will mean people using the APIs will not be affected. We're also going to improve the admin by adding a default `mime` instead of an empty string when we fetch from the DB.

These are in our JIRA board and because we solved the immediate bug you're facing I'm going to close this issue.

If you continue to face these issues with your client i'd recommend you contact support since you're an enterprise customer 😄
